src/test/java/com/fakng/fakngagrgtr/parser/apple/AppleParserTest.java

```diff
@@ -0,0 +1,110 @@
+package com.fakng.fakngagrgtr.parser.apple;
+
+import com.fakng.fakngagrgtr.configuration.ParserConfig;
+import com.fakng.fakngagrgtr.parser.AbstractParserTest;
+import com.fakng.fakngagrgtr.parser.LocationProcessor;
+import com.fakng.fakngagrgtr.persistent.company.Company;
+import com.fakng.fakngagrgtr.persistent.company.CompanyRepository;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import com.gargoylesoftware.htmlunit.WebClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+class AppleParserTest extends AbstractParserTest {
+
+    private static final String URL = "Apple";
+    private AppleParser appleParser;
+    @Mock
+    private WebClient htmlWebClient;
+    @Mock
+    private CompanyRepository companyRepository;
+    @Mock
+    private LocationProcessor locationProcessor;
+
+    @BeforeEach
+    public void init() {
+        appleParser = new AppleParser(
+                htmlWebClient,
+                companyRepository,
+                locationProcessor,
+                URL,
+                new ParserConfig().objectMapper()
```

Comment:
When you write tests, you can create ObjectMapper with "new" operator, because there's no Spring available for you

In fact, what you're doing here is the same. ParserConfig - is also Spring bean that shouldn't be created with "new" operator in usual circumstances. So just create your Object mapper as a field and reuse it. In fact, probably, it will not require changing the bean's configuration in ParserConfig for ignoring properties. Because I don't think that bean should be updated because of tests. Tests should have their own configured dependencies.

Reply:
I've been thinking about it, but I've heard that tests are supposed to check performance. What if I change the ObjectMapper in the configuration, and another implementation remains in the test (because I will create it here via new). Then the tests will remain positive, and the code may stop working due to the new ObjectMapper implementation. Or do I understand something wrong? That's why I wanted to link these 2 points into one.

Reply:
@mighty-mhsl or should the tests depend only on the business logic, and not on the implementation of objects?

Reply:
Unit tests check only the logic of this component working. The tests you are talking about are rather integration tests. We don't write them for now, because they complicate things a lot. Currently, this is not the priority. So a unit test is concentrated only on the logic of the component it's testing. It's called atomicity.